File: Python/232.Implement Queue using Stacks.py
#用栈实现一个队列
class Queue(object):
    def __init__(self):
        self.stack = []

    def push(self, x):
        self.stack.append(x)

    def pop(self):
        if self.stack!=[]:
            self.stack.pop(0)

    def peek(self):
        if self.stack!=[]:
            return self.stack[0]

    def empty(self):
        return self.stack==[]

File: C++/Stack/232.Implement Queue using Stacks.cpp
/*
使用堆栈实现队列。
*/
/*
思路：用两个栈。
 */
#include <iostream>
#include <stack>
using namespace std;

class MyStack {
public:
	stack<int> mystack1;
	stack<int> mystack2; 
    void push(int x) {
        mystack1.push(x);
    }
    
    int pop() 
    {    	
        int res = peek();
        mystack2.pop();
        return res;
    }
    
    int peek() 
    {
        if(mystack2.empty())
        {
        	while(!mystack1.empty())
        	{
        		mystack2.push(mystack1.top());
        		mystack1.pop();
        	}
        }
        return mystack2.top();
    }
    
    bool empty() {        return mystack2.empty() && mystack1.empty();    }
    
};
int main()
{
	MyStack obj;
    obj.push(1);
    cout<< obj.peek() <<endl;
    obj.push(2);
    cout<< obj.pop() <<endl;
    cout<< obj.peek() <<endl;
    cout<< obj.empty() <<endl;
}

